### PR TITLE
Add sphere clustering notebook

### PR DESCRIPTION
**Pull request by johmathe**

### CODE REVIEW

File: notebooks/06_k_mean_clustering_on_sphere.ipynb [hunk 1]
```diff
@@ -0,0 +1,201 @@
+{
```
Comment (ninamiolane) [inline notebooks/06_k_mean_clustering_on_sphere.ipynb:1]: <p>NIT: Import order</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geomstats/geomstats/pull/874/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (ninamiolane) [inline notebooks/06_k_mean_clustering_on_sphere.ipynb:1]: <p>Nice, do you know why the sphere looks like a potato? I had the same issue</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geomstats/geomstats/pull/874/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (ninamiolane) [inline notebooks/06_k_mean_clustering_on_sphere.ipynb:1]: <p>Could add ; at the end of avoid printing the log.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geomstats/geomstats/pull/874/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (cshewmake2) [reply notebooks/06_k_mean_clustering_on_sphere.ipynb:1]: Do you need to set the aspect ratio to equal?

Comment (cshewmake2) [reply notebooks/06_k_mean_clustering_on_sphere.ipynb:1]: https://matplotlib.org/3.1.1/api/_as_gen/matplotlib.axes.Axes.set_aspect.html

Comment (nguigs) [reply notebooks/06_k_mean_clustering_on_sphere.ipynb:1]: <p>This does not work for 3D plots but <code>ax.auto_scale_xyz([-1, 1], [-1, 1], [-1, 1])</code>  seems to do the trick</p>

**Pull request merged**